modules/Kconfig.cmsis_dsp

```diff
@@ -0,0 +1,687 @@
+# Copyright (c) 2020 Stephanos Ioannidis <root@stephanos.io>
```

Comment:
Shouldn't this whole file live in the module?

Reply:
Is there a doc/guideline that specifies how this should be handled?

I see that there are many other modules that have the module configurations in `modules/Kconfig.module`, so I wonder if there is a reason to do things differently for CMSIS-DSP.

Reply:
Haven't looked at the others, but it feels like if the Kconfigs are only used in the module, putting the Kconfig file with the module would make sense to me.

Reply:
To list a few:
https://github.com/zephyrproject-rtos/zephyr/blob/master/modules/Kconfig.mbedtls
https://github.com/zephyrproject-rtos/zephyr/blob/master/modules/Kconfig.tls-generic
https://github.com/zephyrproject-rtos/zephyr/blob/master/modules/Kconfig.tinycbor
https://github.com/zephyrproject-rtos/zephyr/blob/master/modules/Kconfig.tinycrypt

Reply:
I don't think so, for consistency. So far we've avoided adding any Kconfig files in the modules at all, with Kconfig being always in the module. I think we should leave it here, otherwise sooner or later someone will refer to a Kconfig option inside this file and not pull the CMSIS module (i.e. for x86 builds only) and it will fail.

Reply:
> with Kconfig being always in the module

???.  I assume that was a typo.

But good to know that Kconfig isn't in the modules at all.

Reply:
Sorry, yes typo @galak. Edited now.